fastjet/internal/delaunay/triangulation.go

```diff
@@ -0,0 +1,663 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+)
+
+// Delaunay holds necessary information for the
+// delaunay triangulation
+type Delaunay struct {
+	// triangles is a slice of all triangles that have been created. It is used to get the final
+	// list of triangles in the delaunay triangulation
+	triangles triangles
+	// root is a triangle that contains all points. It is used as the starting point in the hierarchy
+	// to find the triangle that contains a point
+	root                   *Triangle
+	maxX, minX, maxY, minY float64
+}
+
+// NewDelaunay creates a delaunay triangulation with the given points
+// all points have to be inside the user defined bounds
+func NewDelaunay(points []*Point, maxX float64, minX float64, maxY float64, minY float64) *Delaunay {
```

Comment:
According to this [paper](https://hal.inria.fr/inria-00166711/document) the time complexity of the walk has a worst case of O(n^5/3). I can add the walk as an option to the user.